HVSample/src/main/java/com/microsoft/hsg/android/hvsample/FileUploadActivity.java

```diff
@@ -36,47 +36,47 @@
 	HealthVaultClient hvClient;
 	
 	private Camera camera;
-    private SurfaceView surfaceView;
-    private View progressContainer;
-    
-    private Camera.ShutterCallback shutterCallback = new Camera.ShutterCallback() {
-        public void onShutter() {
-            // display the progress indicator
-        	progressContainer.setVisibility(View.VISIBLE);
-        }
-    };
-    
-    private Camera.PictureCallback jpegCallback = new Camera.PictureCallback() {
-        public void onPictureTaken(byte[] data, Camera camera) {
-        	
-            String filename = UUID.randomUUID().toString() + ".jpg";
+	private SurfaceView surfaceView;
+	private View progressContainer;
 
-            FileOutputStream os = null;
-            boolean success = true;
-            try {
-                os = openFileOutput(filename, Context.MODE_PRIVATE);
-                os.write(data);
-            } catch (Exception e) {
-                Log.e("FileUpload", "Error writing to file " + filename, e);
-            } finally {
-                try {
-                    if (os != null)
-                        os.close();
-                } catch (Exception e) {
-                    Log.e("FileUpload", "Error closing file " + filename, e);
-                    success = false;
-                } 
-            }
-            
-            if (success) {
-            	com.microsoft.hsg.android.simplexml.things.types.file.File hvFile = new com.microsoft.hsg.android.simplexml.things.types.file.File();
+	private Camera.ShutterCallback shutterCallback = new Camera.ShutterCallback() {
+		public void onShutter() {
+			// display the progress indicator
+			progressContainer.setVisibility(View.VISIBLE);
+		}
+	};
+
+	private Camera.PictureCallback jpegCallback = new Camera.PictureCallback() {
+		public void onPictureTaken(byte[] data, Camera camera) {
+
+			String filename = UUID.randomUUID().toString() + ".jpg";
+
+			FileOutputStream os = null;
+			boolean success = true;
+			try {
+				os = openFileOutput(filename, Context.MODE_PRIVATE);
+				os.write(data);
+			} catch (Exception e) {
+				Log.e("FileUpload", "Error writing to file " + filename, e);
```

Comment:
should set success to false here. Otherwise we can fail to write the file but still have success = true;